### PR TITLE
Added Reconfigure support for a service

### PR DESCRIPTION
**Pull request by h-kataria**

### CODE REVIEW

**Comment by h-kataria** [comment]
Need https://github.com/ManageIQ/manageiq/pull/7073 to be merged to be able to see dialog with filled in fields on ServiceReconfigure request summary screen

screenshots:

Service with reconfigure task available
service_with_reconfigure

Service with no Reconfigure task available
service_with_no_reconfigure

Service Reconfigure form
service_reconfigure

Service Reconfigure form with reconfigurable field
service_reconfigure_with_reconfigurable_fields

after sumbitting reconfigure request
service_after_reconfigure_submit

list of requests showing both types of Requests: Provisioning Request & Reconfigure request
request_list_shwoing_both_types_of_requests

ServiceReconfigure Request summary showing read only dialog
service_reconfigure_request_summary

**Comment by h-kataria** [comment]
@dclarizio @himdel @AparnaKarve ready for review.

File: client/app/states/dashboard/dashboard.state.js [hunk 1]
```diff
@@ -35,34 +35,27 @@
   }
 
   /** @ngInject */
-  function resolvePendingRequests(CollectionsApi, $state) {
-    if (!$state.navFeatures.requests.show) {
-      return undefined;
-    }
-
-    var options = {expand: false, filter: ['approval_state=pending'] };
```
Comment (AparnaKarve) [inline client/app/states/dashboard/dashboard.state.js]: @h-kataria Just wondering why the RBAC checks have been removed here and in the few places below.

Comment (h-kataria) [inline client/app/states/dashboard/dashboard.state.js]: @AparnaKarve that's result of bad merge conflicts, i will take care of those.

**Comment by h-kataria** [comment]
@AparnaKarve fixed code to put the accidentally removed rbac checks back.

File: client/app/states/services/details/details.html [hunk 2]
```diff
@@ -56,8 +56,9 @@
                   <li role="menuitem"><a href="#" ng-click="vm.retireServiceLater()" ng-show="vm.showScheduleRetirementService" translate>Schedule Retirement</a></li>
                 </ul>
               </div>
-              <button class="btn btn-primary" type="button" ng-click="vm.editServiceModal()" ng-show="vm.showEditService" translate>Edit Service</button>
-              <button class="btn btn-primary" type="button" ng-click="vm.ownershipServiceModal()" ng-show="vm.showSetOwnership" translate>Set Ownership</button>
+              <button class="btn btn-primary" type="button" ng-click="vm.editServiceModal()" translate>Edit Service</button>
+              <button class="btn btn-primary" type="button" ng-click="vm.ownershipServiceModal()" translate>Set Ownership</button>
+              <button class="btn btn-primary" type="button" ng-if="vm.service.reconfigure" ng-click="vm.reconfigureService(vm.service.id)" translate>Reconfigure</button>
             </div>
```
Comment (AparnaKarve) [inline client/app/states/services/details/details.html]: @h-kataria Looks like the RBAC check was removed from here as well ...specifically the  ```ng-show``` for ```Edit Service``` and ```Set Ownership``` seems to be deleted and which brings me to the topic of having RBAC support for ```Reconfigure``` -- we would need that as well, right?

Comment (h-kataria) [inline client/app/states/services/details/details.html]: @AparnaKarve fixed, please re-review.

**Comment by h-kataria** [comment]
@AparnaKarve fixed, please re-review.

**Comment by AparnaKarve** [comment]
@dclarizio Tested in UI -- looks good.

OK to merge.

**Pull request merged**